### ISSUE TITLE
"Welcome!" banner text is not localized for non-English languages

### ISSUE THREAD

**Issue by mirunacurtean** [mozilla/sumo-project#809]
**Device:**
Windows 10 64-bit

**Browsers:**
Firefox 86.0 (64-bit)
Chrome Version 89

**Preconditions:**
No account. 

**Steps to reproduce:**

1. Go to a non-English support page https://support.allizom.org/es/
2. Select to Sign In/Up. 
3. Perform sign up process and return to homepage after sign in. 

**Actual result:**
Welcome banner text is not localized in the selected language.
![image](https://user-images.githubusercontent.com/53306802/112133957-e8344780-8bd4-11eb-9d79-fd5ab238a6e6.png)

**Expected results:** 
Welcome banner text is not localized.

**Comment by LeoMcA** [comment]
I think this is happening because the callback URL doesn't have a country-code identifier in it, we'll need to manually set the appropriate locale for strings (we already have some logic to determine this when setting the user's preferred locale in this same route): https://docs.djangoproject.com/en/2.2/topics/i18n/translation/#explicitly-setting-the-active-language

**Comment by mirunacurtean** [comment]
@LeoMcA 
I just tried this while the language was set to ES and then to FR and the welcome banner was still in English. 

![image](https://user-images.githubusercontent.com/53306802/120483722-0e3b4c80-c3bb-11eb-92ff-f34a2bb88434.png)
![image](https://user-images.githubusercontent.com/53306802/120484494-cec13000-c3bb-11eb-9946-c25fc9b90943.png)


In fact, the while sign up process is in English - is that something that only concerns Firefox accounts? 
Another interesting thing to note is that the account language was set to English by default. How is this supposed to work when creating an account while the selected language is non-English? Selecting English by default?
![image](https://user-images.githubusercontent.com/53306802/120483820-2317e000-c3bb-11eb-9a30-d574dd3a3093.png)

**Comment by LeoMcA** [comment]
> In fact, the while sign up process is in English - is that something that only concerns Firefox accounts?

Yeah, FxA takes the language from the Accept-Language header, which is set by the browser.

So what seems to have happened here is:
* You visit fr sumo
* You click sign up
* Your browser is requesting en-US, so FxA displays to you in that locale
* FxA also sets your profile locale to en-US
* On return to sumo, we see that your FxA account is set to en-US, so display the welcome message and set your profile's language to english

If a user's Accept-Language header matches with the sumo locale they're on, things should work as expected.

Still, it's weird and buggy to have a en-US message shown on a generally localized page.

@akatsoulas not sure what the best option here is:
1. Show the welcome message in the locale of the sumo login page, but set the user's profile to the FxA locale
2. Show the welcome message in the locale of the sumo login page, and set the user's profile locale to that as well (ignoring FxA)
3. Show everything in the FxA locale, and set the user's profile to that locale

I think my preference is for 2, possibly 1, since a user is making clear their intent when visiting a localised version of SUMO, and we shouldn't ignore that just because their Accept-Language header doesn't agree.

**Comment by LeoMcA** [comment]
Removing the bug label, and moving this into the blocked column after chatting with tasos about this. While the welcome message appearing in a different locale to the rest of the page is odd, it should only happen when a user's browser locale "disagrees" with the locale they were using on sumo, and it's hard for us to make an assumption about which locale is "correct" in that case.

It might be that we want to display the welcome message in both locales, or we might want to make the user select between the locales. It could also be useful to know how other sites handle this - do they use browser locale, site locale, or even geolocation to determine what language to show (e.g. google.de shows to me in english, is that because of my browser's locale or because google knows I'm in the UK).

We'll need to schedule some time with @dalexsy to go through this at a later date.
